catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/report/RegistryReportBuilder.java

```diff
@@ -0,0 +1,150 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.rest.endpoint.report;
+
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectListWebConverter.ORGANIZATION_KEY;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.common.metacard.RegistryUtility;
+import org.codice.ddf.registry.rest.endpoint.HandlebarsHelper;
+import org.codice.ddf.registry.schemabindings.converter.web.OrganizationWebConverter;
+import org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectListWebConverter;
+import org.codice.ddf.registry.schemabindings.helper.RegistryPackageTypeHelper;
+import org.codice.ddf.registry.schemabindings.helper.WebMapHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+import com.github.jknack.handlebars.helper.StringHelpers;
+import com.github.jknack.handlebars.io.ClassPathTemplateLoader;
+
+import ddf.catalog.data.Metacard;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.OrganizationType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+/**
+ * Generates a specific report as a String of html which is then compiled
+ * and shown through the RegistryRestEndpoint.
+ * <p>
+ * More specifically within this class, maps are generated through the classes contained
+ * in registry-schema-bindings.The details contained in the maps populate a specific handlebar
+ * template and result in a String html that is relayed to RegistryRestEndpoint.
+ */
+public class RegistryReportBuilder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryReportBuilder.class);
+
+    private RegistryObjectListWebConverter registryObjectListWebConverter =
+            new RegistryObjectListWebConverter();
+
+    public String getHTMLFromMetacard(Metacard metacard, ClassPathTemplateLoader templateLoader,
+            String handlebarTemplate) throws IOException {
+
+        Map<String, Object> reportMap = getSummaryMap(metacard);
+        Handlebars handlebars = getHandlebarsFromTemplate(templateLoader);
+        Template template = handlebars.compile(handlebarTemplate);
+        String html = template.apply(reportMap);
+        html = html.replaceAll("\n", "");
```

Comment:
Yeah, still trying to investigate why it is happening. I believe it's because of the loops at the moment